src/scripts/main.js

```diff
@@ -1,3 +1,291 @@
+      const newFieldCells = row.map((value) => {
+        if (value !== 0) {
+          return `<td class="field-cell field-cell--${value}">${value}</td>`;
+        } else {
+          return '<td class="field-cell"></td>';
+        }
+      });
+
+      return `<tr class="field-row">${newFieldCells.join('')}</tr>`;
+    });
+
+    gameField.innerHTML = newFieldRows.join('');
+  }
+
+  goUp() {
+    for (let col = 0; col < 4; col++) {
+      for (let row = 1; row < 4; row++) {
+        if (this.field[row][col] !== 0) {
+          let newRow = row;
+
+          while (newRow > 0
+            && (this.field[newRow - 1][col] === 0
+              || this.field[newRow - 1][col] === this.field[newRow][col])) {
+            if (this.field[newRow - 1][col] === this.field[newRow][col]) {
+              if (this.field[newRow - 1][col] !== 0) {
+                this.field[newRow - 1][col] *= 2;
+                this.field[newRow][col] = 0;
+              }
+            } else if (this.field[newRow - 1][col] === 0) {
+              this.field[newRow - 1][col] = this.field[newRow][col];
+              this.field[newRow][col] = 0;
+            }
+            newRow--;
+          }
+        }
+      }
+    }
+    this.putRandomNumber();
+  }
+
+  goDown() {
+    for (let col = 0; col < 4; col++) {
+      for (let row = 2; row >= 0; row--) {
+        if (this.field[row][col] !== 0) {
+          let newRow = row;
+
+          while (newRow < 3
+            && (this.field[newRow + 1][col] === 0
+              || this.field[newRow + 1][col] === this.field[newRow][col])) {
+            if (this.field[newRow + 1][col] === this.field[newRow][col]) {
+              if (this.field[newRow + 1][col] !== 0) {
+                this.field[newRow + 1][col] *= 2;
+                this.field[newRow][col] = 0;
+              }
+            } else if (this.field[newRow + 1][col] === 0) {
+              this.field[newRow + 1][col] = this.field[newRow][col];
+              this.field[newRow][col] = 0;
+            }
+            newRow++;
+          }
+        }
+      }
+    }
+    this.putRandomNumber();
+  }
+
+  goLeft() {
+    for (let row = 0; row < 4; row++) {
+      for (let col = 1; col < 4; col++) {
+        if (this.field[row][col] !== 0) {
+          let newCol = col;
+
+          while (newCol > 0
+            && (this.field[row][newCol - 1] === 0
+              || this.field[row][newCol - 1] === this.field[row][newCol])) {
+            if (this.field[row][newCol - 1] === this.field[row][newCol]) {
+              if (this.field[row][newCol - 1] !== 0) {
+                this.field[row][newCol - 1] *= 2;
+                this.field[row][newCol] = 0;
+              }
+            } else if (this.field[row][newCol - 1] === 0) {
+              this.field[row][newCol - 1] = this.field[row][newCol];
+              this.field[row][newCol] = 0;
+            }
+            newCol--;
+          }
+        }
+      }
+    }
+    this.putRandomNumber();
+  }
+
+  goRight() {
```

Comment:
there are a lot of duplicated code in methods goUp, goDown, goLeft and goRight
try to improve it